### PR TITLE
Document Preview API

### LINKED ISSUE #417: Document Previews API
We should document this endpoint using a combination of:
- [API Blueprint](https://apiblueprint.org/) for writing the documentation of resources, actions, and URIs
- [Dredd](https://dredd.readthedocs.io/en/latest/) for testing the documentation matches the API
- [Apiary](https://apiary.io/) for browsing the docs

Please reference the journey issue #387 for further instructions or commentary on documenting.

### PR DESCRIPTION
**Pull request by npendery**
Closes #417

### CODE REVIEW

File: blueprint/api.apib [hunk 1]
```diff
@@ -261,6 +285,31 @@ Exchange credentials for `access_token`.
...
+## Preview Response (object)
++ data
+  + id: 1 (string)
+  + type: "previews" (string)
+  + attributes(Preview, required)
+  + relationships(Preview Relationships Base, required)
+
+## Preview Relationship (object)
++ id: 1 (string)
++ type: preview (string)
```
Comment (joshsmith) [inline blueprint/api.apib]: I don't see this used yet. I'm assuming will be coming? The `type` seems different.

File: blueprint/api.apib [hunk 2]
```diff
@@ -261,6 +285,31 @@ Exchange credentials for `access_token`.
 + id: 1 (string)
 + type: organizations (string)
 
+## Preview (object)
++ id: 1 (string)
++ body: "preview" (string)
++ markdown: "#preview" (string)
++ user_id: 1 (string)
```
Comment (joshsmith) [inline blueprint/api.apib]: I would make `markdown` vs `body` a little more explicit here. So more like `A *strong* body`, `A <strong>strong</strong> body`.

Comment (joshsmith) [inline blueprint/api.apib]: Just so it's obvious that `body` is HTML.

File: blueprint/api.apib (hunk 1 above)
Comment (npendery) [inline blueprint/api.apib]: Yeah, there is a relationship with `Preview User Mention`. And I noticed that `preview` should be `previews`, is that what you mean by `type` seems different?

Comment (joshsmith) [inline blueprint/api.apib]: Yeah also quotes and `(string)`.

Comment (npendery) [inline blueprint/api.apib]: Thats how all the other relationship objects are though...sorry, I might be not understanding something?

File: blueprint/api.apib [hunk 3]
```diff
@@ -145,6 +145,30 @@ Exchange credentials for `access_token`.
 
 # Group Previews
 
+## Preview [/categories]
```
Comment (joshsmith) [inline blueprint/api.apib]: Is this supposed to be `/categories`?

File: blueprint/api.apib [hunk 4]
```diff
@@ -145,6 +145,30 @@ Exchange credentials for `access_token`.
...
+  Authorization: Bearer {access_token}
+
++ Response 201 (application/vnd.api+json)
+
+  + Attributes (Preview Response)
+
++ Response 401 (application/json)
+
+    + Attributes (OAuth Invalid Response)
+
```
Comment (joshsmith) [inline blueprint/api.apib:167]: Looks like the `Attributes` indentation is a little inconsistent here.

**Comment by joshsmith** [comment]
Looks good with a couple minor notes.

**Pull request merged**